src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,78 @@
+package core.basesyntax.service;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.model.User;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+
+class RegistrationServiceImplTest {
+    private static StorageDaoImpl storage;
+    private static RegistrationServiceImpl service;
+
+    @BeforeAll
+    static void beforeAll() {
+        storage = new StorageDaoImpl();
+        storage.add(new User(1535854439L, "bobTheOne", "iAmHereAlice", 23));
+        service = new RegistrationServiceImpl();
+    }
+
+    @Test
+    void register_normalUser_Ok() {
+        User normalUser = new User(1535855239L, "MisterSandman", "BringMeADreams", 20);
+        User user = service.register(normalUser);
+        assertNotNull(user);
+        assertEquals(normalUser, user);
+    }
+
+    @Test
+    void register_existingUser_NotOk() {
+        assertThrows(RuntimeException.class, () -> {
+            service.register(new User(1535854439L, "bobTheOne", "iAmHereAlice", 23));
+        });
+    }
+
+    @Test
+    void register_userAgeLessNotValid_NotOk() {
+        assertThrows(RuntimeException.class, () -> {
+            service.register(new User(15358545039L, "Shazam", "SHAZAM!!!", 17));
+        });

```

Comment:
```suggestion
        assertThrows(RuntimeException.class, () -> 
            service.register(new User(15358545039L, "Shazam", "SHAZAM!!!", 17))
        );

```
це можна видалити в усіх тестових методах